src/clj/puppetlabs/services/jruby/jruby_puppet_service.clj

```diff
@@ -32,16 +31,21 @@
         (jruby-agents/send-prime-pool! pool-context)
         (-> context
             (assoc :pool-context pool-context)
-            (assoc :borrow-timeout (:borrow-timeout config))))))
+            (assoc :borrow-timeout (:borrow-timeout config))
+            (assoc :event-callbacks (atom []))))))
 
   (borrow-instance
-    [this]
-    (let [pool-context (:pool-context (tk-services/service-context this))
-          borrow-timeout (:borrow-timeout (tk-services/service-context this))]
-      (core/borrow-from-pool-with-timeout pool-context borrow-timeout)))
+    [this action]
```

Comment:
The `action` seems like something that would be optional.  Should we provide another arity these functions, or perhaps like what we did in jvm-ssl-utils and support an `options` map?